Lib/contextlib.py

```diff
@@ -149,7 +149,10 @@ def __exit__(self, typ, value, traceback):
             except StopIteration:
                 return False
             else:
-                raise RuntimeError("generator didn't stop")
+                try:
+                    raise RuntimeError("generator didn't stop")
+                finally:
+                    self.gen.close()
```

Comment:
👍🏻

I sometimes use the `try:finally:` construct around flow breaking instructions as well. Here it nicely chains exceptions if `self.gen.close()` raises.